lib/memoizable/memory.rb

```diff
@@ -96,8 +97,7 @@ def marshal_dump
     #
     # @api public
     def marshal_load(hash)
-      initialize
-      @memory.marshal_load(hash)
+      initialize(hash)
```

Comment:
__comment:__ This does not seem correct.I think I need to check to see what Hash#maeshal_load does, but I suspect it has semantics similar to if this method was:

```ruby
def marshal_load(hash)
  @monitor.synchronize do
    @memory.replace(hash)
  end
  self
end
```